### PR TITLE
feature: add environment configuration

### PR DESCRIPTION
**Pull request by m-raab**

### CODE REVIEW

**Review by skiesewetter-intershop — CHANGES_REQUESTED** [review]
The general concept of the ENVIRONMENT variable must be discussed.

**Comment by m-raab** [comment]
> The general concept of the ENVIRONMENT variable must be discussed.

I think, as discussed: This variable is supported in ICM. But I think we need an other approach for the configuration, like configuration service.

File: src/main/kotlin/com/intershop/gradle/icm/docker/extension/DevelopmentConfiguration.kt [hunk 1]
```diff
@@ -31,7 +31,7 @@ import org.slf4j.LoggerFactory
 import java.io.File
 import java.io.Serializable
 import java.time.Duration
-import java.util.Properties
+import java.util.*
```
Comment (skiesewetter-intershop) [inline src/main/kotlin/com/intershop/gradle/icm/docker/extension/DevelopmentConfiguration.kt:34]: A  *-import never is a good idea.

File: src/main/kotlin/com/intershop/gradle/icm/docker/extension/DevelopmentConfiguration.kt [hunk 2]
```diff
@@ -256,6 +266,20 @@ open class DevelopmentConfiguration
...
     }
 
+    val intershopEnvironmentProperties: EnvironmentProperties by lazy {
+        val envConfig = EnvironmentProperties(objectFactory)
+        val keys = configProperties.keys.filter {it.toString().startsWith(Configuration.INTERSHOP_ENVIRONMENT_PREFIX)}
+            .stream().map { it.toString()}
+        keys.forEach {
+            val p = getConfigProperty(it)
+            it.replaceFirst(Configuration.INTERSHOP_ENVIRONMENT_PREFIX, ENV_PREFIX)
+                .replace(".", "_")
```
Comment (skiesewetter-intershop) [inline src/main/kotlin/com/intershop/gradle/icm/docker/extension/DevelopmentConfiguration.kt:276]: When providing properties using environment variables starting with `ISH_ENV_` it is not necessary to replace `.` by `_` and converting to upper case.

**Review by skiesewetter-intershop — CHANGES_REQUESTED** [review]

**Review by skiesewetter-intershop — APPROVED** [review]

**Pull request merged**